traits/observation/exception_handling.py

```diff
@@ -40,14 +42,7 @@ def _log_exception(self, event):
         event : object
             An event object emitted by the notification.
         """
-        if self.logger is None:
-            self.logger = logging.getLogger("traits")
-            handler = logging.StreamHandler()
-            handler.setFormatter(logging.Formatter("%(message)s"))
-            self.logger.addHandler(handler)
-            self.logger.setLevel(logging.ERROR)
-
-        self.logger.exception(
+        _logger.exception(
             "Exception occurred in traits notification handler "
             "for event object: %r",
             event,
```

Comment:
The default behaviour of on_trait_change to log to the console stems from an understanding that developers hardly ever configure traits's logger, and any exceptions from the change handlers are likely to be programming errors.
In that case, I think it might actually be better to issue warnings, as they are more visible, so visible that developers will do something about it.